core/services/ardupilot_manager/settings.py

```diff
@@ -12,15 +12,25 @@ class Settings:
     settings_path = Path(appdirs.user_config_dir(app_name))
     settings_file = Path.joinpath(settings_path, "settings.json")
     firmware_path = Path.joinpath(settings_path, "firmware")
+    app_folders = [settings_path, firmware_path]
 
     def __init__(self) -> None:
         self.root: Dict[str, Union[int, Dict[str, Any]]] = {"version": 0, "content": {}}
-
-    def create_settings(self) -> None:
-        """Create settings files and folders."""
+        self.create_app_folders()
+
+    @staticmethod
+    def create_app_folders() -> None:
+        """Create the necessary folders for proper app function."""
+        for folder in Settings.app_folders:
+            try:
+                Path.mkdir(folder, exist_ok=True)
+            except FileExistsError:
+                logger.warning(f"Found file {folder} where a folder should be. Removing file and creating folder.")
```

Comment:
I don't think it is creating the folder after this line. is it?

Reply:
Oh, yes. Good catch. It does not retry.

Reply:
Solved.